pkg/scheduler/framework/frameworkutils.go

```diff
@@ -136,10 +150,13 @@ func crossReferencePickedCustersAndObsoleteBindings(
 		}
 
 		// Update the binding so that it is associated with the lastest scheduling policy.
-		binding.Spec.SchedulingPolicySnapshotName = policy.Name
+		updated.Spec.SchedulingPolicySnapshotName = policy.Name
 
 		// Add the binding to the toUpdate list.
-		toUpdate = append(toUpdate, binding)
+		toPatch = append(toPatch, &bindingWithPatch{
+			updated: updated,
+			patch:   patch,
+		})
```

Comment:
```suggestion
		toPatch = append(toPatch, &bindingWithPatch{
			updated: updated,
			patch:   client.MergeFrom(binding),
		})
```
nit, no need to use the variable or move the variable closer to where it's used